src/api/CqmModelConversionService.ts

```diff
@@ -97,11 +108,147 @@ export class CqmConversionService {
+      population_set_id: `PopulationSet_${i + 1}`,
+      populations: this.generateKeyValuePairPopulations(
+        group.populations,
+        measure.cqlLibraryName
+      ),
+      stratifications: this.generateStratifications(
+        group.stratifications,
+        measure.cqlLibraryName,
+        i
+      ),
+      supplemental_data_elements: this.generateSupplementalDataElements(
+        measure.supplementalData,
+        measure.cqlLibraryName
+      ),
+      ...(measureScoring === "ContinuousVariable" || measureScoring === "Ratio"
+        ? {
+            observations: this.generateObservations(
+              group.measureObservations,
+              measure.cqlLibraryName,
+              group.populations
+            ),
+          }
+        : {}),
+    }));
+
+    return populationSets;
+  };
+
+  private mapPopulationName = (populationName: string) => {
+    if (populationName === PopulationType.INITIAL_POPULATION) return "IPP";
+    if (populationName === PopulationType.DENOMINATOR) return "DENOM";
+    if (populationName === PopulationType.DENOMINATOR_EXCLUSION) return "DENEX";
+    if (populationName === PopulationType.DENOMINATOR_EXCEPTION)
+      return "DENEXCEP";
+    if (populationName === PopulationType.NUMERATOR) return "NUMER";
+    if (populationName === PopulationType.NUMERATOR_EXCLUSION) return "NUMEX";
+    if (populationName === PopulationType.MEASURE_POPULATION) return "MSRPOPL";
+    if (populationName === PopulationType.MEASURE_POPULATION_EXCLUSION)
+      return "MSRPOPLEX";
+    else return populationName;
+  };
+
+  private generateKeyValuePairPopulations = (
+    populations: Population[],
+    cqlLibraryName: string
+  ) => {
+    return populations.reduce((acc, population) => {
+      const key = this.mapPopulationName(population.name);
+      acc[key] = {
+        id: population.id,
+        library_name: cqlLibraryName,
+        statement_name: population.definition,
+        hqmf_id: null,
+      };
+      return acc;
+    }, {});
+  };
+
+  private generateObservations = (
+    observations: MeasureObservation[],
+    cqlLibraryName: string,
+    populations: Population[]
+  ) => {
+    return observations.map((observation, i) => ({
+      id: observation.id,
+      hqmf_id: null,
+      aggregation_type: observation.aggregateMethod,
+      observation_function: {
+        id: uuidv4(),
+        library_name: cqlLibraryName,
+        statement_name: observation.definition,
+        hqmf_id: null,
+      },
+      observation_parameter: {
+        id: uuidv4(),
+        library_name: cqlLibraryName,
+        statement_name: this.getObservationAssocationName(
+          observation.criteriaReference,
+          populations
+        ),
+        hqmf_id: null,
+      },
+    }));
+  };
+
+  private getObservationAssocationName = (
+    criteriaReference: string,
+    populations: Population[]
+  ) => {
+    populations.filter((population) => population.id === criteriaReference)[0]
+      ?.name;
```

Comment:
don't think this is being used?

Reply:
My bad, I forgot to remove some unwanted code I'll make the changes